plugins/woocommerce/client/admin/client/task-lists/fills/PaymentGatewaySuggestions/index.js

```diff
@@ -72,39 +71,46 @@ export const PaymentGatewaySuggestions = ( { onComplete, query } ) => {
 		[ installedPaymentGateways, paymentGatewaySuggestions ]
 	);
 
-	const enablePaymentGateway = ( id ) => {
-		if ( ! id ) {
-			return;
-		}
+	const enablePaymentGateway = useCallback(
+		( id ) => {
+			if ( ! id ) {
+				return;
+			}
 
-		const gateway = getPaymentGateway( id );
+			const gateway = getPaymentGateway( id );
 
-		if ( ! gateway ) {
-			return;
-		}
+			if ( ! gateway ) {
+				return;
+			}
 
-		updatePaymentGateway( id, {
-			enabled: true,
-		} ).then( () => {
-			onComplete(
-				// use the paymentGateways variable.
-				// gateway variable doesn't have hasPlugins property.
-				! paymentGateways.get( id )?.hasPlugins
-					? {
-							// If we are already on a task page, don't redirect.
-							// Otherwise, redirect to Payments task page.
-							redirectPath: getQuery()?.task
-								? getNewPath(
-										{ task: getQuery().task },
-										{},
-										'/'
-								  )
-								: getNewPath( { task: 'payments' }, {}, '/' ),
-					  }
-					: {}
-			);
-		} );
-	};
+			updatePaymentGateway( id, {
+				enabled: true,
+			} ).then( () => {
+				onComplete(
+					// use the paymentGateways variable.
+					// gateway variable doesn't have hasPlugins property.
+					! paymentGateways.get( id )?.hasPlugins
+						? {
+								// If we are already on a task page, don't redirect.
+								// Otherwise, redirect to Payments task page.
+								redirectPath: getQuery()?.task
+									? getNewPath(
+											{ task: getQuery().task },
```

Comment:
I feel like I'm lost in a maze here. Not your code, just felt like sharing my feelings 😄 
![its-a-trap-gif-by-star-wars](https://github.com/user-attachments/assets/a8605a14-ba4b-4261-9eb2-4bf65bf0312f)

Reply:
Hehe, that's how I've been feeling the whole time working on this 😅